### PR TITLE
Use govet plugin instead of maligned

### PR DESCRIPTION
**Pull request by shahpratikr**
## Change Overview

https://github.com/mdempsky/maligned is deprecated.
maligned | The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
Hence changed maligned plugin to govet.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Review by pavannd1 — DISMISSED** [review]

File: build/golint.sh [hunk 1]
```diff
@@ -22,7 +22,9 @@ TIMEOUT="10m"
 
 echo "Running golangci-lint..."
 
-golangci-lint run --timeout ${TIMEOUT} --skip-dirs ${SKIP_DIR_REGEX} -E maligned,whitespace,gocognit,unparam -e '`ctx` is unused'
+# https://github.com/mdempsky/maligned is deprecated.
+# maligned | The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
```
Comment (julio-lopez) [inline build/golint.sh:26]: This comment is only helpful in the context of this change. It belongs in the PR description. It adds noise going forward here in the script, please remove.
```suggestion
```

File: build/golint.sh [hunk 2]
```diff
@@ -22,7 +22,9 @@ TIMEOUT="10m"
...
+golangci-lint run --timeout ${TIMEOUT} --skip-dirs ${SKIP_DIR_REGEX} -E govet,whitespace,gocognit,unparam -e '`ctx` is unused'
```
Comment (julio-lopez) [inline build/golint.sh:27]: (Un?)related question, not requesting a change. Why isn't a configuration file used for these runs?

Comment (shahpratikr) [reply build/golint.sh:27]: there is https://github.com/kanisterio/kanister/pull/1921 this PR which adds configuration file. But not sure about what's pending there.

File: build/golint.sh (hunk 1 above)
Comment (shahpratikr) [reply build/golint.sh]: fixed

**Review by julio-lopez — DISMISSED** [review]

**Comment by julio-lopez** [comment]
@shahpratikr Please rebase onto `master` just the changes corresponding `golangci-lint`
I'd be happy to approve after that.

**Comment by shahpratikr** [comment]
@julio-lopez rebased with master. changed base branch to master as well.

**Pull request closed without merging**